### PR TITLE
initial dps service client split

### PR DESCRIPTION
**Pull request by jasmineymlo**
- changed service client netstandard project file to the new format
- added provisioning service client implementation

### CODE REVIEW

File: provisioning/service/src/DeviceRegistrationNotFoundException.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+using Microsoft.Azure.Devices.Common;
+
+namespace Microsoft.Azure.Devices.Provisioning.Service
+{
+    using System;
+    using System.Runtime.Serialization;
+    using Microsoft.Azure.Devices.Common.Exceptions;
+
+#if !WINDOWS_UWP
+    [Serializable]
```
Comment (ryan-z-johnson) [inline provisioning/service/src/DeviceRegistrationNotFoundException.cs]: Is this not valid on UWP? Do we have an option to do one that is? something like [this?](https://stackoverflow.com/questions/32446989/system-serializable-attribute-gone-in-windows-10-uwp-apps)

File: provisioning/service/src/DeviceRegistrationOperationError.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    public sealed class DeviceRegistrationOperationError
+    {
+        [JsonProperty(PropertyName = "registrationId")]
+        public string RegistrationId { get; set; }
+
+        [JsonProperty(PropertyName = "errorCode")]
+        public string ErrorCode { get; set; }
+
+        [JsonProperty(PropertyName = "errorStatus")]
```
Comment (ryan-z-johnson) [inline provisioning/service/src/DeviceRegistrationOperationError.cs]: Can we not have the JsonProperty have the same capitalization as the class or is this defined by the api?

File: provisioning/service/src/HttpDeviceRegistrationClient.cs [hunk 3]
```diff
@@ -0,0 +1,660 @@
...
+            {
+                if (this.httpClientHelper != null)
+                {
+                    this.httpClientHelper.Dispose();
+                    this.httpClientHelper = null;
+                }
+            }
+        }
+
+        void EnsureInstanceNotClosed()
```
Comment (CIPop) [inline provisioning/service/src/HttpDeviceRegistrationClient.cs]: Naming is confusing: maybe `ThrowIfClosed()`?

File: provisioning/service/src/HttpDeviceRegistrationClient.cs [hunk 4]
```diff
@@ -0,0 +1,660 @@
...
+
+        public override Task<Enrollment> GetEnrollmentAsync(string registrationId)
+        {
+            return this.GetEnrollmentAsync(registrationId, CancellationToken.None);
+        }
+
+        public override Task<Enrollment> GetEnrollmentAsync(string registrationId, CancellationToken cancellationToken)
+        {
+            this.EnsureInstanceNotClosed();
+            var errorMappingOverrides = new Dictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>>
```
Comment (CIPop) [inline provisioning/service/src/HttpDeviceRegistrationClient.cs]: Perf: This seems to be quite heavy to do each request. Instead of creation a dictionary every time, a bit-mask might be faster.

File: provisioning/service/src/HttpDeviceRegistrationClient.cs [hunk 5]
```diff
@@ -0,0 +1,660 @@
...
+            {
+                {
+                    HttpStatusCode.NotFound,
+                    responseMessage => Task.FromResult((Exception) new EnrollmentNotFoundException(registrationId))
```
Comment (CIPop) [inline provisioning/service/src/HttpDeviceRegistrationClient.cs]: Perf: Allocating this exception beforehand is guaranteeing that we always consume the memory. Instead, we could allocate only when we need to throw that exception.

File: provisioning/service/src/HttpDeviceRegistrationClient.cs [hunk 6]
```diff
@@ -0,0 +1,660 @@
...
+        }
+
+        public override Task<Enrollment> UpdateEnrollmentAsync(Enrollment enrollment, bool forceUpdate)
+        {
+            return this.UpdateEnrollmentAsync(enrollment, forceUpdate, CancellationToken.None);
+        }
+
+        public override Task<Enrollment> UpdateEnrollmentAsync(Enrollment enrollment, bool forceUpdate, CancellationToken cancellationToken)
+        {
+            this.EnsureInstanceNotClosed();
```
Comment (CIPop) [inline provisioning/service/src/HttpDeviceRegistrationClient.cs]: We should remove all `this.` following the coding guideline.
Current code isn't consistent either (see below).

File: provisioning/service/src/HttpDeviceRegistrationClient.cs [hunk 7]
```diff
@@ -0,0 +1,660 @@
...
+        public override Task<BulkOperationResult> UpdateEnrollmentsAsync(IEnumerable<Enrollment> enrollments, bool forceUpdate)
+        {
+            return this.UpdateEnrollmentsAsync(enrollments, forceUpdate, CancellationToken.None);
+        }
+
+        public override Task<BulkOperationResult> UpdateEnrollmentsAsync(IEnumerable<Enrollment> enrollments, bool forceUpdate, CancellationToken cancellationToken)
+        {
+            this.EnsureInstanceNotClosed();
+
+            foreach (Enrollment enrollment in enrollments)
```
Comment (CIPop) [inline provisioning/service/src/HttpDeviceRegistrationClient.cs]: Duplicate code: we could extract a method and use this wherever we need to validate an `IEnumerable<Enrollment>` parameter.

File: provisioning/service/src/HttpDeviceRegistrationClient.cs [hunk 8]
```diff
@@ -0,0 +1,660 @@
...
+            }
+
+            if(!string.IsNullOrWhiteSpace(enrollment.DeviceId) &&
+               !DeviceIdRegex.IsMatch(enrollment.DeviceId))
+            {
+                throw new ArgumentException(ApiResources.EnrollmentIdNotValid.FormatInvariant(enrollment.DeviceId));
+            }
+        }
+
+        static void Validate(EnrollmentGroup enrollmentGroup)
```
Comment (CIPop) [inline provisioning/service/src/HttpDeviceRegistrationClient.cs]: missing `internal`
Also the name is ambiguous. `ValidateEnrollmentGroup` maybe?

File: provisioning/service/src/HttpDeviceRegistrationClient.cs [hunk 9]
```diff
@@ -0,0 +1,660 @@
...
+        const string QueryEnrollmentGroupUriFormat = "enrollmentGroups/query?{0}";
+        const string DeviceRegistrationUriFormat = "registrations/{0}?{1}";
+        const string DeviceRegistrationsUriFormat = "registrations/{0}/query?{1}";
+
+        const string ContinuationTokenHeader = "x-ms-continuation";
+        const string PageSizeHeader = "x-ms-max-item-count";
+
+        private const string ApiVersionQueryString = ClientApiVersionHelper.ApiVersionQueryString;
+
+        static readonly Regex DeviceIdRegex = new Regex(@"^[A-Za-z0-9\-:.+%_#*?!(),=@;$']{1,128}$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
```
Comment (CIPop) [inline provisioning/service/src/HttpDeviceRegistrationClient.cs]: Naming is confusing: `DeviceAndRegistrationIdRegex` or simply `IDRegex` maybe?

File: provisioning/service/src/HttpDeviceRegistrationClient.cs [hunk 10]
```diff
@@ -0,0 +1,660 @@
...
+            {
+                throw new ArgumentException("enrollmentGroup.EnrollmentGroupId");
+            }
+        }
+
+        async Task<DpsQueryResult> ExecuteQueryAsync(Uri queryUri, int? pageSize, string continuationToken, CancellationToken cancellationToken)
+        {
+            this.EnsureInstanceNotClosed();
+
+            //todo: support sqlQueryStrings
```
Comment (CIPop) [inline provisioning/service/src/HttpDeviceRegistrationClient.cs]: Sounds like we should add a GH issue and reference it here for these kinds of TODOs so we don't forget we have them.

File: provisioning/service/src/Microsoft.Azure.Devices.Provisioning.Service.csproj [hunk 11]
```diff
@@ -2,13 +2,67 @@
 
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
   </PropertyGroup>
 
   <ItemGroup>
+    <Compile Include="..\..\..\common\src\service\AssertionFailedException.cs" Link="common\AssertionFailedException.cs" />
```
Comment (CIPop) [inline provisioning/service/src/Microsoft.Azure.Devices.Provisioning.Service.csproj]: TODO: We need to create a common include csproj which defines a $(Common) that we can use throughout our projects.

File: provisioning/service/src/Microsoft.Azure.Devices.Provisioning.Service.csproj [hunk 12]
```diff
@@ -2,13 +2,67 @@
...
+    <PackageReference Include="Newtonsoft.Json" Version="9.0.1" />
+  </ItemGroup>
+
+  <ItemGroup>
     <ProjectReference Include="..\..\..\shared\Microsoft.Azure.Devices.Shared.NetStandard\Microsoft.Azure.Devices.Shared.NetStandard.csproj">
       <Project>{91dfb837-d8a3-4f54-ae0d-45c95acd0c2a}</Project>
       <Name>Microsoft.Azure.Devices.Shared.NetStandard</Name>
     </ProjectReference>
+    <ProjectReference Include="..\..\..\service\Microsoft.Azure.Devices.NetStandard\Microsoft.Azure.Devices.NetStandard.csproj" />
+    <ProjectReference Include="..\..\..\shared\Microsoft.Azure.Devices.Shared.NetStandard\Microsoft.Azure.Devices.Shared.NetStandard.csproj" />
```
Comment (CIPop) [inline provisioning/service/src/Microsoft.Azure.Devices.Provisioning.Service.csproj]: Duplicate: please remove.

File: provisioning/service/src/Properties/Resources.Designer.cs [hunk 13]
```diff
@@ -0,0 +1,63 @@
+﻿//------------------------------------------------------------------------------
```
Comment (CIPop) [inline provisioning/service/src/Properties/Resources.Designer.cs]: Do we need this and the Resources.resx file as well as the 2 above?

File: service/Microsoft.Azure.Devices.NetStandard/Microsoft.Azure.Devices.NetStandard.csproj [hunk 14]
```diff
@@ -218,15 +261,10 @@
     <Compile Include="..\Microsoft.Azure.Devices\ServiceStatistics.cs" />
     <Compile Include="..\Microsoft.Azure.Devices\SymmetricKey.cs" />
     <Compile Include="..\Microsoft.Azure.Devices\X509Thumbprint.cs" />
-    <Compile Include="Properties\AssemblyInfo.cs" />
   </ItemGroup>
   <ItemGroup>
-    <ProjectReference Include="..\..\shared\Microsoft.Azure.Devices.Shared.NetStandard\Microsoft.Azure.Devices.Shared.NetStandard.csproj">
-      <Project>{91dfb837-d8a3-4f54-ae0d-45c95acd0c2a}</Project>
```
Comment (CIPop) [inline service/Microsoft.Azure.Devices.NetStandard/Microsoft.Azure.Devices.NetStandard.csproj]: We should leave the GUID and name in place.

File: service/Microsoft.Azure.Devices.NetStandard/Properties/AssemblyInfo.cs [hunk 15]
```diff
@@ -40,3 +40,4 @@
 // Version information for an assembly follows semantic versioning 1.0.0 (because
 // NuGet didn't support semver 2.0.0 before VS 2015). See semver.org for details.
 [assembly: AssemblyInformationalVersion("1.3.2")]
+[assembly: InternalsVisibleTo("Microsoft.Azure.Devices.Provisioning.Service")]
```
Comment (CIPop) [inline service/Microsoft.Azure.Devices.NetStandard/Properties/AssemblyInfo.cs]: We need to perform the split w/o the need of `InternalsVisibleTo`.

File: service/Microsoft.Azure.Devices/QueryResponse.cs [hunk 16]
```diff
@@ -14,7 +14,7 @@ public class QueryResponse<T> : IEnumerable<T>
     {
         readonly IEnumerable<T> queryResults;
 
-        internal QueryResponse(IEnumerable<T> queryResults, string continuationToken)
+        public QueryResponse(IEnumerable<T> queryResults, string continuationToken)
```
Comment (CIPop) [inline service/Microsoft.Azure.Devices/QueryResponse.cs]: This is a public API change (note that we will need to increase the ver number).

File: service/iothub_csharp_serviceclient.sln [hunk 17]
```diff
@@ -303,6 +303,7 @@ Global
 		{C1F0833C-C4BE-4DC8-9B3B-71283E0E2F35} = {E26199F8-3D1F-4BC0-A70A-8597EA9865A0}
 	EndGlobalSection
 	GlobalSection(ExtensibilityGlobals) = postSolution
+		SolutionGuid = {5791B3ED-C5DB-470D-B987-49AE9D92EF44}
```
Comment (CIPop) [inline service/iothub_csharp_serviceclient.sln]: Why was this added? Is there a new project added to this solution?

File: service/iothub_csharp_serviceclient.sln [hunk 18]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 14
-VisualStudioVersion = 14.0.25420.1
+# Visual Studio 15
```
Comment (CIPop) [inline service/iothub_csharp_serviceclient.sln]: Is this necessary to build the new netstandard project file?

File: service/Microsoft.Azure.Devices/QueryResponse.cs (hunk 16 above)
Comment (ryan-z-johnson) [reply service/Microsoft.Azure.Devices/QueryResponse.cs]: Can I jump onto this and ask why we are moving it from internal to public?

File: provisioning/service/src/DeviceRegistrationNotFoundException.cs [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+#endif
+    public sealed class DeviceRegistrationNotFoundException : IotHubException
```
[the remark this answers is not in the document]
Comment (jasmineymlo) [reply provisioning/service/src/DeviceRegistrationNotFoundException.cs]: Maybe we should create a new class for ProvisioningException?

---
In reply to: [141717282](https://github.com/Azure/azure-iot-sdk-csharp/pull/174#discussion_r141717282) [](ancestors = 141717282)

File: service/Microsoft.Azure.Devices/QueryResponse.cs (hunk 16 above)
Comment (jasmineymlo) [reply service/Microsoft.Azure.Devices/QueryResponse.cs]: agree to up the version of service client

---
In reply to: [141728780](https://github.com/Azure/azure-iot-sdk-csharp/pull/174#discussion_r141728780) [](ancestors = 141728780)

Comment (jasmineymlo) [reply service/Microsoft.Azure.Devices/QueryResponse.cs]: because this class is referenced by both iothub and dps service clients.  In order for dps service clients to referenced it in the iothub project, we have to expose it...

---
In reply to: [141730370](https://github.com/Azure/azure-iot-sdk-csharp/pull/174#discussion_r141730370) [](ancestors = 141730370)

File: provisioning/service/src/HttpDeviceRegistrationClient.cs (hunk 10 above)
Comment (jasmineymlo) [reply provisioning/service/src/HttpDeviceRegistrationClient.cs]: created Task 1580139 to track

---
In reply to: [141726374](https://github.com/Azure/azure-iot-sdk-csharp/pull/174#discussion_r141726374) [](ancestors = 141726374)

File: provisioning/service/src/Enrollment.cs [hunk 20]
```diff
@@ -0,0 +1,87 @@
...
+        /// </summary>
+        [JsonConverter(typeof(StringEnumConverter))]
+        [JsonProperty(PropertyName = "provisioningStatus", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public ProvisioningStatus? ProvisioningStatus { get; set; }
+
+        /// <summary>
+        /// The DateTime this resource was created.
+        /// </summary>
+        [JsonProperty(PropertyName = "createdDateTimeUtc", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public DateTime? CreatedDateTimeUtc { get; set; }
```
[the remark this answers is not in the document]
Comment (jasmineymlo) [reply provisioning/service/src/Enrollment.cs]: Task 1582640 to track

---
In reply to: [141718625](https://github.com/Azure/azure-iot-sdk-csharp/pull/174#discussion_r141718625) [](ancestors = 141718625)

File: provisioning/service/src/EnrollmentGroup.cs [hunk 21]
```diff
@@ -0,0 +1,75 @@
...
+        /// </summary>
+        [JsonConverter(typeof(StringEnumConverter))]
+        [JsonProperty(PropertyName = "provisioningStatus", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public ProvisioningStatus? ProvisioningStatus { get; set; }
+
+        /// <summary>
+        /// The DateTime this resource was created.
+        /// </summary>
+        [JsonProperty(PropertyName = "createdDateTimeUtc", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public DateTime? CreatedDateTimeUtc { get; set; }
```
[the remark this answers is not in the document]
Comment (jasmineymlo) [reply provisioning/service/src/EnrollmentGroup.cs]: Task 1582640 to track

---
In reply to: [141719415](https://github.com/Azure/azure-iot-sdk-csharp/pull/174#discussion_r141719415) [](ancestors = 141719415)

File: provisioning/service/src/DeviceRegistrationNotFoundException.cs (hunk 19 above)
[the remark this answers is not in the document]
Comment (jasmineymlo) [reply provisioning/service/src/DeviceRegistrationNotFoundException.cs]: Since it is already in Provisioing.Service namespace.  I have removed "Device" to make it clearer.

---
In reply to: [141717282](https://github.com/Azure/azure-iot-sdk-csharp/pull/174#discussion_r141717282) [](ancestors = 141717282)

File: common/src/service/ExceptionTrace.cs [hunk 22]
```diff
@@ -214,13 +213,13 @@ internal void BreakOnException(Exception exception)
                     {
                         Debugger.Launch();
                     }
-#elif !PCL && !NETSTANDARD1_3
+#elif !PCL && !NETSTANDARD1_3 && !NETSTANDARD2_0
```
[the remark this answers is not in the document]
Comment (jasmineymlo) [reply common/src/service/ExceptionTrace.cs]: NET45 is not defined in the older csproj.  Will keep it as-is for now.  Ifdef cleanup will happen when netstandard1.3 release is official so that we could remove support for packages for individual platform.

---
In reply to: [141714190](https://github.com/Azure/azure-iot-sdk-csharp/pull/174#discussion_r141714190) [](ancestors = 141714190)

File: common/src/service/Fx.cs [hunk 23]
```diff
@@ -441,7 +441,7 @@ internal static bool FastDebug
         static bool TryGetDebugSwitch(string name, out object value)
         {
             value = null;
-#if !WINDOWS_UWP && !NETSTANDARD1_3
+#if !WINDOWS_UWP && !NETSTANDARD1_3 && !NETSTANDARD2_0
```
[the remark this answers is not in the document]
Comment (jasmineymlo) [reply common/src/service/Fx.cs]: see above

---
In reply to: [141714284](https://github.com/Azure/azure-iot-sdk-csharp/pull/174#discussion_r141714284) [](ancestors = 141714284)

File: service/Microsoft.Azure.Devices/QueryResponse.cs (hunk 16 above)
Comment (jasmineymlo) [reply service/Microsoft.Azure.Devices/QueryResponse.cs]: We move it to internal to public because the class is now shared between iothub service client and provisioning service client.  We have plans to refactor these commons into the shared.

File: provisioning/service/src/AttestationMechanism.cs [hunk 24]
```diff
@@ -36,10 +36,10 @@ public AttestationMechanism()
         [JsonProperty(PropertyName = "tpm")]
         public TpmAttestation Tpm
         {
-            get { return this.tpm; }
+            get { return this._tpm; }
```
Comment (CIPop) [inline provisioning/service/src/AttestationMechanism.cs]: We should remove the `this.` part. "_" means that it's a local property already.

File: provisioning/service/src/HttpDeviceRegistrationClient.cs (hunk 4 above)
Comment (jasmineymlo) [reply provisioning/service/src/HttpDeviceRegistrationClient.cs]: #176 to track

---
In reply to: [141721788](https://github.com/Azure/azure-iot-sdk-csharp/pull/174#discussion_r141721788) [](ancestors = 141721788)

File: provisioning/service/src/HttpDeviceRegistrationClient.cs (hunk 5 above)
Comment (jasmineymlo) [reply provisioning/service/src/HttpDeviceRegistrationClient.cs]: #176 to track

---
In reply to: [141722179](https://github.com/Azure/azure-iot-sdk-csharp/pull/174#discussion_r141722179) [](ancestors = 141722179)

File: provisioning/service/src/HttpDeviceRegistrationClient.cs [hunk 25]
```diff
@@ -0,0 +1,660 @@
...
+            this.httpClientHelper = new HttpClientHelper(
+                connectionString.HttpsEndpoint,
+                connectionString,
+                ExceptionHandlingHelper.GetDefaultErrorMapping(),
+                DefaultOperationTimeout,
+                client => { });
+        }
+
+        // internal test helper
+        internal HttpDeviceRegistrationClient(IHttpClientHelper httpClientHelper, string drsName)
```
[the remark this answers is not in the document]
Comment (jasmineymlo) [reply provisioning/service/src/HttpDeviceRegistrationClient.cs]: removed and we will need to figure out another way to test this instead of migrating the exisiting tests over.

---
In reply to: [141720270](https://github.com/Azure/azure-iot-sdk-csharp/pull/174#discussion_r141720270) [](ancestors = 141720270)

File: provisioning/service/src/HttpDeviceRegistrationClient.cs [hunk 26]
```diff
@@ -0,0 +1,660 @@
...
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Shared;
+    using Microsoft.Azure.Devices.Common;
+    using Microsoft.Azure.Devices.Common.Exceptions;
+
+    class HttpDeviceRegistrationClient : DeviceRegistrationClient, IDisposable
```
[the remark this answers is not in the document]
Comment (jasmineymlo) [reply provisioning/service/src/HttpDeviceRegistrationClient.cs]: #176 to track

---
In reply to: [141719853](https://github.com/Azure/azure-iot-sdk-csharp/pull/174#discussion_r141719853) [](ancestors = 141719853)

File: provisioning/service/src/DeviceRegistrationClient.cs [hunk 27]
```diff
@@ -0,0 +1,386 @@
...
+    using System.Collections.Generic;
+    using System.Net.Http.Headers;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Shared;
+    
+    /// <summary>
+    /// Contains methods that services can use to perform create, remove, update and delete operations on devices.
+    /// </summary>
+    public abstract class DeviceRegistrationClient : IDisposable
```
[the remark this answers is not in the document]
Comment (jasmineymlo) [reply provisioning/service/src/DeviceRegistrationClient.cs]: #176 to track

---
In reply to: [141716525](https://github.com/Azure/azure-iot-sdk-csharp/pull/174#discussion_r141716525) [](ancestors = 141716525)

File: provisioning/service/src/HttpDeviceRegistrationClient.cs (hunk 26 above)
[the remark this answers is not in the document]
Comment (jasmineymlo) [reply provisioning/service/src/HttpDeviceRegistrationClient.cs]: #176 to track

---
In reply to: [141719957](https://github.com/Azure/azure-iot-sdk-csharp/pull/174#discussion_r141719957) [](ancestors = 141719957)

File: common/src/service/ExceptionTrace.cs (hunk 22 above)
[the remark this answers is not in the document]
Comment (CIPop) [reply common/src/service/ExceptionTrace.cs]: Tracked by #176.

File: common/src/service/Resources.Designer.cs [hunk 28]
```diff
@@ -0,0 +1,506 @@
...
+//     Runtime Version:4.0.30319.42000
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+using System.Reflection;
+
+namespace Microsoft.Azure.Devices.Common {
```
Comment (CIPop) [inline common/src/service/Resources.Designer.cs]: We've discussed about this offline: we should remove localization.
Tracked by #177.

/cc @jasmineymlo @tameraw

File: provisioning/service/src/HttpProvisioningServiceClient.cs [hunk 29]
```diff
@@ -0,0 +1,655 @@
...
+            {
+                throw new ArgumentException("enrollmentGroup.EnrollmentGroupId");
+            }
+        }
+
+        async Task<ProvisioningQueryResult> ExecuteQueryAsync(Uri queryUri, int? pageSize, string continuationToken, CancellationToken cancellationToken)
+        {
+            ThrowIfClosed();
+
+            //TODO: support sqlQueryStrings
```
Comment (CIPop) [inline provisioning/service/src/HttpProvisioningServiceClient.cs]: Tracked by #176

File: provisioning/service/src/Properties/ApiResources.Designer.cs [hunk 30]
```diff
@@ -0,0 +1,522 @@
...
+// <auto-generated>
+//     This code was generated by a tool.
+//     Runtime Version:4.0.30319.42000
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Devices {
```
Comment (CIPop) [inline provisioning/service/src/Properties/ApiResources.Designer.cs]: Tracked by #177
